allauth/socialaccount/providers/lichess/views.py

```diff
@@ -0,0 +1,64 @@
+from allauth.socialaccount import app_settings
+from allauth.socialaccount.adapter import get_adapter
+from allauth.socialaccount.providers.oauth2.views import (
+    OAuth2Adapter,
+    OAuth2CallbackView,
+    OAuth2LoginView,
+)
+
+
+class LichessOAuth2Adapter(OAuth2Adapter):
+    provider_id = "lichess"
+
+    settings = app_settings.PROVIDERS.get(provider_id, {})
+    provider_base_url = settings.get("API_URL", "https://lichess.org")
+
+    access_token_url = "{0}/api/token".format(provider_base_url)
+    authorize_url = "{0}/oauth".format(provider_base_url)
+
+    profile_url = "{0}/api/account".format(provider_base_url)
+    email_address_url = "{0}/api/account/email".format(provider_base_url)
+
+    def complete_login(self, request, app, token, response):
+        extra_data = (
+            get_adapter()
+            .get_requests_session()
+            .get(
+                self.profile_url,
+                params={"access_token": token.token},
+                headers={
+                    "Authorization": f"Bearer {token.token}",
+                },
+            )
+        )
+
+        # retrieve email address
+        email_data = (
```

Comment:
This is an additional call that is only needed if this is true:

```
from allauth.socialaccount.app_settings import QUERY_EMAIL
```

Can you make that optional?